hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -602,6 +603,10 @@ void Vgen::emit(const andli& i) {
   a->And(W(i.d), W(i.s1), MSKTOP(i.s0.l()), SetFlags);
 }
 
+void Vgen::emit(const testbi& i) {
+  a->Tst(W(i.s1), MSKTOP(i.s0.l()));
+}
```

Comment:
Why not implement this inline with the other `test*{}` instructions?

I'm also still not sure I understand the semantics of `MSKTOP()` and why you're using `i.s0.l()` for byte instructions when you could be using `i.s0.b()`—but this is a more general question, not specific to this instruction in particular.  @dave-estes, can you explain in more detail?